pkg/controller/kogitoapp/resource/build_config_s2i.go

```diff
@@ -117,6 +120,21 @@ func setBCS2IStrategy(kogitoApp *v1alpha1.KogitoApp, buildConfig *buildv1.BuildC
 		envs = framework.EnvOverride(envs, corev1.EnvVar{Name: mavenMirrorURLEnvVar, Value: kogitoApp.Spec.Build.MavenMirrorURL})
 	}
 
+	if len(kogitoApp.Spec.Build.Artifact.GroupId) > 0 {
```

Comment:
I would add a extra if here checking if the build is really from file and a build using the archetype needed.

Reply:
Actually there are two options when artifact configuration can be user:
- build from file
- build from URL pointing to file or dir with source file

Reply:
If URL is supplied we have no way of identifying where does it lead to. This type of checks are already implemented in builder images where we checkout complete git repo from provided url and verify whether build using the archetype needed. Currently not implementing any checks here

Reply:
don't you agree that you are setting unneeded things parameters?